### PR TITLE
Alignment of delete_batch_metadata function

### PR DESCRIPTION
**Pull request by mdbitz**
Addition of post-trashed status and cleanup of cases. Introduces a new reset_deleted_metadata_mapping that can be used by extending classes to have custom cache deletion logic.

#### Changes proposed in this Pull Request:
* N/A - Cleanup of technical debt.

#### Does this pull request change what data or activity we track or use?
- NO

#### Testing instructions:

* On Master perform the following:
* Create several new posts that have a meta entry for new key "my-key".
* Open wp shell and execute
* `$store = new \Automattic\Jetpack\Sync\Replicastore();`
* `$store->delete_batch_metadata('post', [ Post IDs ], "my-key");
* Verify the meta was removed.
* Apply this PR / checkout branch
* Repeat the same steps

#### Proposed changelog entry for your changes:
* Jetpack Sync : Update of Replicastore functions with enhancements from WordPress.com implementation.

### CODE REVIEW

**Review by anomiex — CHANGES_REQUESTED** [review]

File: packages/sync/src/class-replicastore.php [hunk 1]
```diff
@@ -720,16 +719,31 @@ public function delete_batch_metadata( $type, $object_ids, $meta_key ) {
...
 	}
 
+	/**
+	 * Delete meta cache of an object.
+	 *
+	 * @param string $type      Meta type.
+	 * @param int    $object_id ID of the object.
+	 * @return void
+	 */
+	private function reset_deleted_metadata_mapping( $type, $object_id ) {
```
[the remark this answers is not in the document]
Comment (mdbitz) [reply packages/sync/src/class-replicastore.php:743]: Thanks, oversight on my end I was simply pulling the diff across vs thinking through usage. I should know better than to code while helping the kids with school :)

File: packages/sync/src/class-replicastore.php [hunk 2]
```diff
@@ -720,16 +719,31 @@ public function delete_batch_metadata( $type, $object_ids, $meta_key ) {
 		if ( ! $table ) {
 			return false;
 		}
-		$column = sanitize_key( $type . '_id' );
+		$meta_key             = esc_html( $meta_key );
```
[the remark this answers is not in the document]
Comment (mdbitz) [reply packages/sync/src/class-replicastore.php:722]: We allow for users to extend the meta allow list to any value. This is a precaution we needed to take on WP.com as it was causing issues during various down stream functionality. As this doesn't interact with site data there is no impact to remote sites, instead if a meta key had html in it they would simply be stored with a modified key on the replicastore.

[the remark this answers is not in the document]
Comment (anomiex) [reply packages/sync/src/class-replicastore.php:720]: I'd still say that escaping before inserting into the DB is an antipattern. But I don't know anything about those downstream uses, so I won't try to block on it.

I note that if there are any existing meta keys that actually will get escaped, this won't delete existing unescaped versions anymore.

**Comment by mdbitz** [comment]
Thinking about this more and the down stream implementation can be further modified. Going to close for now and revisit.

**Pull request closed without merging**